### PR TITLE
Make to_camel_case and capitalize_first configurable

### PR DESCRIPTION
**Pull request by jordangarside**
Adds `auto_camelcase` and `auto_capitalize` bools inside of `str_converters` that can be modified to stop camelCasing or Capitalizing.

```py
from strawberry.utils import str_converters

str_converters.auto_camelcase = False
```

I wasn't sure of a better way to do this, as these utilities are used in various functions that can be run at any time before a schema is generated.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* https://github.com/strawberry-graphql/strawberry/issues/542

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

There wasn't any test on these methods already and I don't think this functionality is very complicated at the moment.
Also not sure about adding to the docs since it might not be the long-term API that is wanted here.

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: strawberry/utils/str_converters.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
+auto_camelcase = True
```
Comment (BryceBeagle) [inline strawberry/utils/str_converters.py:1]: Nitpick: We're being inconsistent about whether whether it's `camel_case` or `camelcase`

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

**Comment by BryceBeagle** [comment]
Are global switches the best option? I think in the ideal situation, this configuration should be stored in the (each) `Schema` object, but that may be more work than its worth.

**Comment by patrick91** [comment]
> Are global switches the best option? I think in the ideal situation, this configuration should be stored in the (each) Schema object, but that may be more work than its worth.

I'd prefer this too if possible, might be a bit of pain to implement though 🤔

File: strawberry/utils/str_converters.py (hunk 1 above)
Comment (jordangarside) [reply strawberry/utils/str_converters.py:1]: Yeah I was debating that here. Graphene had the inconsistency so I was keeping it the same as that. I'll update it.

**Comment by AndrewIngram** [comment]
> > Are global switches the best option? I think in the ideal situation, this configuration should be stored in the (each) Schema object, but that may be more work than its worth.
> 
> I'd prefer this too if possible, might be a bit of pain to implement though 🤔

@patrick91 What in particular makes it a pain? Is it the problem of essentially having to pass around the schema/config to all the locations that may need it?

**Comment by patrick91** [comment]
> @patrick91 What in particular makes it a pain? Is it the problem of essentially having to pass around the schema/config to all the locations that may need it?

@AndrewIngram yes :) also it kinda prevents to convert a type without having a schema (ie if you want to see how a type looks like in the SDL)

**Comment by AndrewIngram** [comment]
Another use case for ContextVars? https://docs.python.org/3/library/contextvars.html

**Comment by jordangarside** [comment]
The issue here is that you need to set `auto_camel_case` before you use any of the `field` or `type` functions in strawberry. So adding the option to `strawberry.Schema` won't actually affect the behavior (even if you use a contextVar there), because the name has already been camelCased before you even initialize the schema.

i.e. In the way strawberry is currently setup, you have to modify _some_ global value before you run any strawberry methods to make a feature like this work.

**Pull request closed without merging**